http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcSecurityRealmTest.java

```diff
@@ -0,0 +1,187 @@
+import org.wildfly.security.authz.Attributes;
+import org.wildfly.security.authz.AuthorizationIdentity;
+
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+import static org.wildfly.security.auth.SupportLevel.POSSIBLY_SUPPORTED;
+import static org.wildfly.security.auth.SupportLevel.UNSUPPORTED;
+import static org.wildfly.security.auth.server.RealmIdentity.NON_EXISTENT;
+import static org.wildfly.security.authz.RoleDecoder.KEY_ROLES;
+
+/**
+ * Tests the {@link OidcSecurityRealm} implementation.
+ *
+ * @author <a href="mailto:patrick@reini.net">Patrick Reinhart</a>
+ */
+public class OidcSecurityRealmTest {
+    private OidcSecurityRealm realm;
+
+    @Before
+    public void setUp() {
+        realm = new OidcSecurityRealm();
+    }
+
+    @Test
+    public void testGetCredentialAcquireSupport() throws RealmUnavailableException {
+        assertEquals(UNSUPPORTED, realm.getCredentialAcquireSupport(null, null, null));
+    }
+
+    @Test
+    public void testGetEvidenceVerifySupport() throws RealmUnavailableException {
+        assertEquals(POSSIBLY_SUPPORTED, realm.getEvidenceVerifySupport(null, null));
+    }
+
+    @Test
+    public void testGetRealmIdentityWithNonOidcPrincipal() throws RealmUnavailableException {
+        Principal nonOidcPricipal = new SimplePrincipal("john");
+        assertEquals(NON_EXISTENT, realm.getRealmIdentity(nonOidcPricipal));
+    }
+
+    @Test
+    public void testGetRealmIdentityNoRoles() throws RealmUnavailableException {
+        // setup
+        RefreshableOidcSecurityContext securityContext = new RefreshableOidcSecurityContext();
+        securityContext.setCurrentRequestInfo(new OidcClientConfiguration(), null);
+        OidcPrincipal principal = new OidcPrincipal("john", securityContext);
+
+        // test
+        RealmIdentity identity = realm.getRealmIdentity(principal);
+
+        // verification
+        assertNotEquals(NON_EXISTENT, identity);
+        assertEquals(principal, identity.getRealmIdentityPrincipal());
+        assertEquals(UNSUPPORTED, identity.getCredentialAcquireSupport(null, null, null));
+        assertNull(identity.getCredential(null));
+        assertEquals(SupportLevel.SUPPORTED, identity.getEvidenceVerifySupport(null, null));
+        assertTrue(identity.verifyEvidence(null));
+        assertTrue(identity.exists());
+        AuthorizationIdentity authorizationIdentity = identity.getAuthorizationIdentity();
+        assertNotNull(authorizationIdentity);
+        final Attributes.Entry roles = authorizationIdentity.getAttributes().get(KEY_ROLES);
+        assertTrue(roles.isEmpty());
+    }
+
+    @Test
+    public void testGetRealmIdentityRolesCombined() throws RealmUnavailableException {
+        // setup
+        final OidcClientConfiguration clientConfiguration = new OidcClientConfiguration();
+        clientConfiguration.setResource("SpecialResource");
```

Comment:
It would be better to use `setClientId` instead of `setResource` (`resource` is terminology from the Keycloak adapter, the OIDC spec uses `clientId` instead).